makeSpectral.py

```diff
@@ -0,0 +1,259 @@
+		dt = self.swit(dt, 'std::make_shared<BaseParameters>', f'std::make_shared<{self.name}Parameters>')
+		dt = self.swit(dt, 'std::make_unique<BaseProcessor>', f'std::make_unique<{self.name}Processor>')
+		dt = self.swit(dt, 'std::shared_ptr<BaseParameters>', f'std::shared_ptr<{self.name}Parameters>')
+		return dt
+
+class AudioPluggin_H(Jucer):
+	@override
+	def manip(self, dt : str) -> str:
+		dt = self.swit(dt, '#include "BaseProcessor.h"', f'#include "{self.name}Processor.h"')
+		dt = self.swit(dt, '#include "BaseParameters.h"', f'#include "{self.name}Parameters.h"')
+		return dt
+
+class Parameters_Cpp(Jucer):
+	@override
+	def manip(self, dt : str) -> str:
+		dt = self.swit(dt, '#include "BaseParameters.h"', f'#include "{self.name}Parameters.h"')
+		dt = self.swit(dt, 
+			'BaseParameters::BaseParameters(AudioProcessor * processor)', 
+			f'{self.name}Parameters::{self.name}Parameters(AudioProcessor * processor)')
+		return dt
+
+class Parameters_H(Jucer):
+	@override
+	def manip(self, dt : str) -> str:
+		dt = self.swit(dt, 
+			'BaseParameters(AudioProcessor* processor);', 
+			f'{self.name}Parameters(AudioProcessor* processor);')
+		dt = self.swit(dt, 
+			'class BaseParameters : public PluginParameters', 
+			f'class {self.name}Parameters : public PluginParameters')
+		return dt
+
+class Processor_Cpp(Jucer):
+	@override
+	def manip(self, dt : str) -> str:
+		dt = self.swit(dt, '#include "BaseProcessor.h"', f'#include "{self.name}Processor.h"')
+		dt = self.swit(dt, 
+			'std::unique_ptr<StandardFFTProcessor> BaseProcessor', 
+			f'std::unique_ptr<StandardFFTProcessor>{self.name}Processor')
+		dt = self.swit(dt, 
+			'void BaseProcessor::prepareProcess(StandardFFTProcessor * spectralProcessor)', 
+			f'void {self.name}Processor::prepareProcess(StandardFFTProcessor * spectralProcessor)')
+		return dt
+
+class Processor_H(Jucer):
+	@override
+	def manip(self, dt : str) -> str:
+		dt = self.swit(dt, '#include "BaseParameters.h"', f'#include "{self.name}Parameters.h"')
+		dt = self.swit(dt, 
+			'class BaseProcessor : public SpectralAudioProcessorInteractor',
+			f'class {self.name}Processor : public SpectralAudioProcessorInteractor')
+		dt = self.swit(dt, 
+			'BaseProcessor(int numOverlaps, std::shared_ptr<BaseParameters> params)', 
+			f'{self.name}Processor(int numOverlaps, std::shared_ptr<BaseParameters> params)')
+		dt = self.swit(dt, 'std::shared_ptr<BaseParameters> m_params', f'std::shared_ptr<{self.name}Parameters> m_params')
+		return dt
+
+class UiContainer_Cpp(Jucer):
+	@override
+	def manip(self, dt : str) -> str:
+		dt = self.swit(dt, 
+			'std::shared_ptr<BaseParameters> valueTreeState', 
+			f'std::shared_ptr<{self.name}Parameters> valueTreeState')
+		return dt
+
+	@override
+	def rename_path(self, old_path : str) ->  str:
+		return old_path
+
+class UiContainer_H(Jucer):
+	@override
+	def manip(self, dt : str) -> str:
+		dt = self.swit(dt, '#include "BaseParameters.h"', f'#include "{self.name}Parameters.h"')
```

Comment:
This is really nice. We might have to ammend these classes if we change the Base files, but it's safer being explicit on these. Good idea and very readable